### PR TITLE
Rails example

### PR DESCRIPTION
**Pull request by grubernaut**
Terraform modules, documentation, and Atlas testing need to be completed
and finalized.

Committing now, to ensure packer builds AMI's correctly in Atlas before
building out terraform infrastructure and finalizing documentation.

### CODE REVIEW

**Comment by KFishner** [comment]
Now that the Rails stuff is split out, you can remove those files

**Comment by grubernaut** [comment]
App moved to github.com/hashicorp/demo-app-rails

File: rails/packer/aws/ubuntu/base.json [hunk 1]
```diff
@@ -0,0 +1,75 @@
+{
+    "variables": {
+        "aws_access_key":   "{{env `AWS_ACCESS_KEY`}}",
+        "aws_secret_key":   "{{env `AWS_SECRET_KEY`}}",
+        "atlas_username":   "{{env `ATLAS_USERNAME`}}",
```
Comment (sethvargo) [inline rails/packer/aws/ubuntu/base.json]: I think this is the wrong envvar. There are a few username variables, but I don't think this is one. I think you might want `ATLAS_BUILD_CONFIGURATION_USERNAME`?

File: rails/packer/config/consul/rails.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+    "service": {
```
Comment (sethvargo) [inline rails/packer/config/consul/rails.json]: I don't like tabs in json, but that's just me :smile:

File: rails/packer/scripts/ubuntu/consul.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+set -e
+
+cd /tmp
+
+CONSULURL=https://releases.hashicorp.com/consul/0.6.0-rc1/consul_0.6.0-rc1_linux_amd64.zip
+CONSULUI=https://releases.hashicorp.com/consul/0.6.0-rc1/consul_0.6.0-rc1_web_ui.zip
```
Comment (sethvargo) [inline rails/packer/scripts/ubuntu/consul.sh]: rc2 is out now

File: rails/packer/scripts/ubuntu/consul.sh [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+CONSULCONFIGDIR=/etc/consul.d
+CONSULDIR=/opt/consul
+
+curl -L $CONSULURL > /tmp/consul.zip
+
+unzip /tmp/consul.zip -d /usr/local/bin
```
Comment (sethvargo) [inline rails/packer/scripts/ubuntu/consul.sh]: I'm not seeing where we've installed unzip?

File: rails/packer/aws/ubuntu/base.json (hunk 1 above)
Comment (grubernaut) [inline rails/packer/aws/ubuntu/base.json]: I think we specify `ATLAS_USERNAME` for atlas examples in order to use `packer push` when no configuration is present.
https://github.com/hashicorp/atlas-examples/blob/master/letschat/packer/nodejs.json#L3
https://github.com/hashicorp/atlas-examples#environment-variables

File: rails/packer/scripts/ubuntu/deps.sh [hunk 5]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+set -e
+
+apt-get -y update
```
Comment (sethvargo) [inline rails/packer/scripts/ubuntu/deps.sh]: I think you're going to need the cloud-init nonsense here: https://github.com/sethvargo/terraform-workshop-atlas/blob/5e154ceae0ecb0eb81ed95ea7d45cc318c548e6d/05-atlas/terraform/scripts/wait-for-ready.sh

File: rails/packer/scripts/ubuntu/consul.sh (hunk 4 above)
Comment (grubernaut) [inline rails/packer/scripts/ubuntu/consul.sh]: https://github.com/hashicorp/atlas-examples/pull/80/files#diff-dae19d1b049c7dc3b716215e9f2d5745R6

File: rails/packer/scripts/ubuntu/deps.sh (hunk 5 above)
Comment (grubernaut) [inline rails/packer/scripts/ubuntu/deps.sh]: Yeah, good point thanks :)

File: rails/packer/aws/ubuntu/base.json (hunk 1 above)
Comment (sethvargo) [inline rails/packer/aws/ubuntu/base.json]: Ah - so it's an examples-specific thing. :+1:

File: rails/packer/config/consul/rails.json (hunk 2 above)
Comment (grubernaut) [inline rails/packer/config/consul/rails.json]: It's spaces, but `tab-width` was set to 4. Changing to 2 :)

File: rails/packer/aws/ubuntu/base.json (hunk 1 above)
Comment (KFishner) [inline rails/packer/aws/ubuntu/base.json]: Seth is right, we should use `ATLAS_BUILD_CONFIGURATION_USERNAME`. The previous setup with `ATLAS_USERNAME` was a hack before we had that option.

Comment (grubernaut) [inline rails/packer/aws/ubuntu/base.json]: Ah awesome, will fix thanks

**Comment by grubernaut** [comment]
@sethvargo @KFishner updated based on comments

File: rails/packer/aws/ubuntu/base.json [hunk 6]
```diff
@@ -0,0 +1,80 @@
+{
+  "variables": {
+    "aws_access_key": "{{env `AWS_ACCESS_KEY`}}",
+    "aws_secret_key": "{{env `AWS_SECRET_KEY`}}",
```
Comment (clstokes) [inline rails/packer/aws/ubuntu/base.json]: Also, best to use `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` which are the common AWS ones that AWS-CLI, Terraform, and other tools expect.

Comment (grubernaut) [inline rails/packer/aws/ubuntu/base.json]: Updated

**Comment by KFishner** [comment]
LGTM

**Pull request merged**